devicetypes/smartthings/zigbee-ceiling-fan-light.src/led-fan-lightings.groovy

```diff
@@ -0,0 +1,245 @@
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		main(["switch"])
+		details(["switch", "refresh", "switchLevel"])
+	}
+}
+
+private getFAN_CLUSTER_VALUE() { 0x0202 }
+private getFAN_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+
+def parse(String description) {
+	// Parse incoming device messages to generate events
+	log.debug "description is $description"
+	def event = zigbee.getEvent(description)
+	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	if (event) {
+		sendEvent(event)
+	} else if (description?.startsWith('read attr -')) {
+		if (zigbeeMap.clusterInt == FAN_CLUSTER_VALUE &&
+		    zigbeeMap.attrInt == FAN_STATUS_VALUE) {		
+			log.debug "read attribute event for fan cluster attrib FAN_STATUS"
+			def childDevice = getChildDevices()?.find {
+				//find light child device
+				log.debug "parse() child device found"
+				it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+			}          
+			event.displayed = true
+			zigbeeMap.name = "fanSpeed"
+			childDevice.sendEvent(zigbeeMap)
+			if (zigbeeMap.value == "00") {
+				log.debug "fan_off => switch off"
+				childDevice.sendEvent(name: "switch", value: "off")
+				childDevice.sendEvent(name: "level", value: 0)	// For cloud to cloud device UI update
+			} else {
+				log.debug "fan_on => switch on"
+				childDevice.sendEvent(name: "switch", value: "on")
+				def int_v  = zigbeeMap.value as Integer
+				int_v = int_v * 25
+				int_v = int_v > 100 ? 100 : int_v
+				log.debug "child device Level set $int_v"
+				childDevice.sendEvent(name: "level", value: int_v)	// For cloud to cloud device UI update
+			}
+		}		
+	} else {
+		def cluster = zigbee.parse(description)
+		if (cluster && 
+		    cluster.clusterId == ON_OFF_CLUSTER_VALUE &&
+		    cluster.command == 0x07) {
+			if (cluster.data[0] == 0x00) {
+				log.debug "ON/OFF REPORTING CONFIG RESPONSE: " + cluster
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value, rate = null, device = null) {
+	if (device == null) {
+		zigbee.setLevel(value)
+	} else {
+    		if (value <= 1) {
+            		setFanSpeed(0, device)
+        	} else if (value <= 25) {
+            		setFanSpeed(1, device)
+        	} else if (value <= 50) {
+            		setFanSpeed(2, device)
+        	} else if (value <= 75) {
+            		setFanSpeed(3, device)
+        	} else if (value <= 100) {
+            		setFanSpeed(4, device)
+        	}        
```

Comment:
So referring to discussion here https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/68445/files#r670393336, will this part of code be actually ever reached? It isn't called from the child device, and I highly doubt that argument `device` will be ever different than `null`, as it even not considered as a capability's command argument.

Reply:
That code is a function overloading. It can separates what is called between a child or a main.

Reply:
Walk me through it, cause I don't see it. Child DTH calls only parent's `setFanSpeed`.

Reply:
What a cool programmer you are! Yes, you're right. Thanks for looking closely. This part uses for cloud to cloud connection. I told before that there is not fan speed capability in other platforms. So we have to use switch level capability instead of fan speed capability. Then other platform user can see the switch level UI for control fan speed. When the user change the switch level for fan speed, child DTH calls parent's setFanSpeed. And that point of the time SmartThings hub operates for fan speed control.

Reply:
Are you able to tell me which 3rd party cloud you want to integrate it with? 
Will it call SmartThings API? 
Have you tested communication between ST and 3rd party with this current DTH?
Right now I doubt it'd work, cause `device` isn't listed as command's attribute https://smartthings.developer.samsung.com/docs/api-ref/capabilities.html#Switch-Level

Reply:
![image](https://user-images.githubusercontent.com/74279632/126581346-87a715c4-01fe-480d-b62d-8007c1c23df7.png)

Reply:
![image](https://user-images.githubusercontent.com/74279632/127978037-cdad118b-8c02-4880-85fd-f6c397bb2443.png)

Reply:
Sorry for late answer. I needed to change code to use capability "Switch Level" for control fan speed in other platforms. I upload a image file. I captured some screens when I use different platforms. You can see that. Those are google home and amazon alexa.

Reply:
Move this logic to to `setLevel()` you've created in the child DTH.

Reply:
If I move that logic to child DTH, dimming level and fan speed change together.